### PR TITLE
Fix unnormalized small-x Planck integral

### LINKED ISSUE #2069: Small-x Planck integral is not normalized
> *This was generated by AI during triage.*

## Summary
The analytic small-x branch returns the unnormalized Planck integral even though the function contract and interpolation table return a fraction of the total integral.

## Severity
`High`

## Affected File
`src/radiation/planck_integral.hpp`

## Affected Function / Symbol
`integrate_planck_from_0_to_x(...)`

## Audit Metadata
- Source log: `issues/new/planck-small-x-integral-is-not-normalized.md`
- Finding tags: correctness/radiation/numerics

## Proposed Patch
Evaluate a cancellation-free small-x series, such as `x^3/3 - x^4/8 + x^5/60`, and divide it by `gInf`. Match the series continuously to the interpolation table without a monotonicity clamp. Add logarithmically spaced accuracy and monotonicity tests spanning zero, the branch boundary, and the saturated large-x regime.

## Why This Is a Bug
`integrate_planck_from_0_to_x()` promises the ratio to `pi^4 / 15`. For `log10(x) < -3`, it evaluates an approximation to the raw integral but never divides by `gInf`. Around `x = 5e-4`, the result is therefore about 6.49 times too large. The subsequent clamp to `Y_INTERP_MIN` creates an artificial plateau before the table boundary. At still smaller x, subtractive cancellation in `(-4+x)*x + 8*log((2+x)/2)` causes even larger relative errors.

This function feeds radiation group Planck fractions and group-edge derivatives, so sufficiently small dimensionless band edges receive incorrect weights.

### PR DESCRIPTION
**Pull request by chongchonghe**
### Description
`integrate_planck_from_0_to_x()` (`src/radiation/planck_integral.hpp`) is meant to return the Planck integral fraction, normalized by `gInf = π^4/15`. Its small-x branch (`x < 1e-3`) had two problems:

1. **Missing normalization:** it returned the raw unnormalized series — about `gInf ≈ 6.49x` too large — which fed incorrect group weights into small multigroup RHD group boundaries.
2. **Incorrect series:** the closed form `(-4 + x)·x + 8·ln((2 + x)/2)` is only accurate through 4th order. Its Taylor expansion is `x³/3 − x⁴/8 + x⁵/20 − …`, but the true integral of `t³/(eᵗ−1)` expands to `x³/3 − x⁴/8 + x⁵/60 − …` (the x⁵ coefficient is 1/60, not 1/20).

Fix: replace the small-x branch with the correct 5th-order Taylor series `(x³/3 − x⁴/8 + x⁵/60) / gInf`, normalized by `gInf` to match the interpolation-table branch. For `x < 1e-3` the x⁵ term is negligible, so numerical results are unchanged, but the expression is now mathematically correct to 5th order.

### Related issues
Closes #2069

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [x] I have added a description (see above).
- [x] I have added a link to any related issues (if applicable; see above).
- [x] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [ ] I have added tests for any new physics that this PR adds to the code. (N/A — bug fix, no new physics; validated with the existing multigroup `RadTube` test, see Validation below.)
- [ ] *(For quokka-astro org members)* I have manually triggered the GPU tests with the magic comment `/azp run`.

### Validation
Built and ran the multigroup RHD `RadTube` test (2 groups) under the `1d` preset:
```
Relative L1 norm = 6.827103469e-05
RadTube SUCCESS
```
(tolerance `rel_err_tol = 0.003` in `testRadTube.cpp`)

Regression check with a related, unmodified multigroup test, `RadhydroPulseMGconst` (`1d` preset):
```
Relative L1 error norm = 0.004194961044
RadhydroPulseMGconst SUCCESS
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Planck integral accuracy for very small input values by updating the small‑x approximation.
  * Enhanced alignment between approximation results and interpolated computations.
  * Kept the existing monotonicity behavior and validation checks unchanged to maintain expected output consistency.

### CODE REVIEW

File: src/radiation/planck_integral.hpp [hunk 1]
```diff
@@ -243,7 +243,7 @@ AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE auto integrate_planck_from_0_to_x(const
 	Real y = NAN;
 	if (logx < LOG_X_MIN) {
 		// y = x * x * x / 3.0;    // 1st order
-		y = (-4 + x) * x + 8 * std::log((2 + x) / 2); // 2nd order
+		y = ((-4 + x) * x + 8 * std::log((2 + x) / 2)) / gInf; // 2nd order, normalized by gInf to match interpolate_planck_integral()
```
[the remark this answers is not in the document]
Comment (BenWibking) [reply src/radiation/planck_integral.hpp]: This is what Codex says:
```
It suffers severe subtractive cancellation: the order-(x) and order-(x^2) terms cancel, leaving an order-(x^3) result.

  In double precision:

  - (x=10^{-4}): about 0.25% relative error
  - (x=10^{-5}): about 79% relative error
  - (x=10^{-6}): about 1,678× too large
  - Smaller values can become negative and get incorrectly clamped to zero

  std::log1p(x / 2) helps, but cancellation remains. Use the cancellation-free small-(x) expansion:

  const Real x2 = x * x;
  y = x2 * x * (1.0 / 3.0 + x * (-1.0 / 8.0 + x / 60.0)) / gInf;

  This approximates

  [
  \int_0^x \frac{t^3}{e^t-1},dt
  = \frac{x^3}{3}-\frac{x^4}{8}+\frac{x^5}{60}+O(x^7),
  ]

  and is essentially full double-precision accuracy throughout the branch (0 < x < 10^{-3}).

  Note also that expanding the current logarithmic expression gives an (x^5/20) term, not the correct (x^5/60).
  ```

**Comment by BenWibking** [comment]
I asked Codex for a LaTeX derivation, and it gave me this: [planck_small_x_expansion.pdf](https://github.com/user-attachments/files/30190144/planck_small_x_expansion.pdf).

How did you compute your small-x approximation?

**Comment by chongchonghe** [comment]
> I asked Codex for a LaTeX derivation, and it gave me this: [planck_small_x_expansion.pdf](https://github.com/user-attachments/files/30190144/planck_small_x_expansion.pdf).
> 
> How did you compute your small-x approximation?

That was a dumb approximate algebra I did. My formula is correct to the fourth order, but wrong to the fifth order. There is really no reason to use the more expensive `std::log` over the cheaper and more accurate Taylor series. Dumb. I'll correct it. 
CleanShot 2026-07-21 at 10 47 26@2x

**Comment by chongchonghe** [comment]
@BenWibking Ready

**Review by BenWibking — APPROVED** [review]
